storage/storage_transport_test.go

```diff
@@ -0,0 +1,137 @@
+
+		{"UPPERCASEISINVALID@" + sha256digestHex, "", ""},                                                                  // Invalid name in name@ID
+		{"busybox@ab", "", ""},                                                                                             // Invalid ID in name@ID
+		{"busybox@", "", ""},                                                                                               // Empty ID in name@ID
+		{"busybox@sha256:" + sha256digestHex, "", ""},                                                                      // This (in a digested docker/docker reference format) is also invalid
+		{"@" + sha256digestHex, "", sha256digestHex},                                                                       // Valid two-component name, with ID only
+		{"busybox@" + sha256digestHex, "docker.io/library/busybox:latest", sha256digestHex},                                // Valid two-component name, implicit tag
+		{"busybox:notlatest@" + sha256digestHex, "docker.io/library/busybox:notlatest", sha256digestHex},                   // Valid two-component name, explicit tag
+		{"docker.io/library/busybox:notlatest@" + sha256digestHex, "docker.io/library/busybox:notlatest", sha256digestHex}, // Valid two-component name, everything explicit
+	} {
+		storageRef, err := Transport.ParseStoreReference(Transport.(*storageTransport).store, c.input)
+		if c.expectedRef == "" && c.expectedID == "" {
+			assert.Error(t, err, c.input)
+		} else {
+			require.NoError(t, err, c.input)
+			assert.Equal(t, *(Transport.(*storageTransport)), storageRef.transport, c.input)
+			assert.Equal(t, c.expectedRef, storageRef.reference, c.input)
+			assert.Equal(t, c.expectedID, storageRef.id, c.input)
+			if c.expectedRef == "" {
+				assert.Nil(t, storageRef.name, c.input)
+			} else {
+				dockerRef, err := reference.ParseNamed(c.expectedRef)
+				require.NoError(t, err)
+				require.NotNil(t, storageRef.name, c.input)
+				assert.Equal(t, dockerRef.String(), storageRef.name.String())
+			}
+		}
+	}
+}
+
+func TestTransportParseReference(t *testing.T) {
+	store := newStore(t)
+	driver := store.GetGraphDriverName()
+	root := store.GetGraphRoot()
+
+	for _, c := range []struct{ prefix, expectedDriver, expectedRoot string }{
+		{"", driver, root},          // Implicit store location prefix
+		{"[unterminated", "", ""},   // Unterminated store specifier
+		{"[]", "", ""},              // Empty store specifier
+		{"[relative/path]", "", ""}, // Non-absolute graph root path
+		//{"[" + root + "suffix1]", driver, root + "suffix1"}, // A valid root path FIXME: this currently fails
+		{"[" + driver + "@relative/path]", "", ""},      // Non-absolute graph root path
+		{"[thisisunknown@" + root + "suffix2]", "", ""}, // Unknown graph driver
+		//{"[" + driver + "@" + root + "suffix3]", driver, root + "suffix3"}, // A valid root@graph  FIXME: this currently fails
```

Comment:
Reworded.  Any mitigation the storage library might provide for common cases later, by caching the settings used for a given graph directory, wouldn't work for the test because the test always creates a unique directory to use.